cli/ubi.go

```diff
@@ -71,9 +71,14 @@ func sendRequest(args []string) {
 		fmt.Fprintf(os.Stderr, "! Error sending http request\n")
 		os.Exit(1)
 	}
-	defer resp.Body.Close()
 
 	processResponse(resp, args)
+
+	err = resp.Body.Close()
```

Comment:
Should we defer to close in case it exits before reaching this point? Something like that

```golang
defer func() {
    err := resp.Body.Close()
    if err != nil {
        fmt.Fprintf(os.Stderr, "! Error closing response body\n")
 		os.Exit(1)
    }
}()
```

Reply:
There are no return statements that can cause early exits between the `defer` call and the end of the function, so I don't think the `defer` call is actually necessary.  That's why I moved it to the end of the function.  However, I'm certainly not a go expert.  Is there is a way for `processResponse` to cause an early exit of the function?

Reply:
It has some `os.Exit(1)` calls, but I haven't been writing Go for +5 years, so I'm not sure if it's an issue or not.

Reply:
`os.Exit` does not call deferred functions in Go, so it wouldn't have been called anyway: https://github.com/golang/go/issues/38261

Even if it would have been called, it wouldn't matter in this case, because process exit implicitly closes file descriptors.